src/core/channels/encrypted_channel.ts

```diff
@@ -32,7 +32,7 @@ export default class EncryptedChannel extends PrivateChannel {
       if (!sharedSecret) {
         let errorMsg = `No shared_secret key in auth payload for encrypted channel: ${this.name}`;
         callback(true, errorMsg);
-        Logger.warn(`Error: ${errorMsg}`);
+        Logger.error(errorMsg);
```

Comment:
Is this necessary now? We log the error in the callback anyway?

Reply:
Isn't the `callback` passed in from the user of the library? The authorizer of the superclass is called further up in this method.

Reply:
I don't think so. The `callback` is passed in by the `subscribe` method of the `Channel` type here:

https://github.com/pusher/pusher-js/blob/error-logging-improvements/src/core/channels/channel.ts#L97

This calls `authorize` on the `EncryptedChannel` which called `authorize` on the `PrivateChannel` etc

Reply:
Ah yes you are correct. I found the flow confusing going from `Channel.subscribe -> EncryptedChannel.authorize -> PrivateChannel.authorize`. Good catch!